dotnet/src/Experimental/Process.Abstractions/KernelProcessFunctionTarget.cs

```diff
@@ -10,14 +10,15 @@ public record KernelProcessFunctionTarget
     /// <summary>
     /// Creates an instance of the <see cref="KernelProcessFunctionTarget"/> class.
     /// </summary>
-    public KernelProcessFunctionTarget(string stepId, string functionName, string? parameterName = null)
+    public KernelProcessFunctionTarget(string stepId, string functionName, string? parameterName = null, string? targetEventId = null)
     {
         Verify.NotNullOrWhiteSpace(stepId);
         Verify.NotNullOrWhiteSpace(functionName);
 
         this.StepId = stepId;
         this.FunctionName = functionName;
         this.ParameterName = parameterName;
+        this.TargetEventId = targetEventId;
```

Comment:
I still don't understand why properties being assigned in the constructor are declare as `init`.  Are they intended to be assigned twice?